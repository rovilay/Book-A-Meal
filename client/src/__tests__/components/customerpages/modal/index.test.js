import React from 'react';
import { shallow } from 'enzyme';
import { orderDetails } from '../../../__mockData__/orderMock';
import ModalComp from '../../../../components/customerpages/modal/Index';

describe('ModalComp component test', () => {
  const setup = () => {
    const props = {
      modal: {
        content: {
          ...orderDetails,
          createdAt: '2018-09-01'
        },
        contentLabel: 'Edit Order',
        isOpen: true,
        isInfo: true,
        isEdit: true
      },
      getOrderMeals: jest.fn(),
      title: 'Order Details',
      content: {
        ...orderDetails,
        createdAt: '2018-09-01'
      },
      orderedMealsPagination: {
        limit: 12,
        count: 13,
        numOfPages: 2
      },
      setModal: jest.fn(),
      hideModal: jest.fn(),
      editOrder: jest.fn()
    };

    return shallow(<ModalComp {...props} />)
  };

  it('should render `ModalComp` correctly', (done) => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });
});
