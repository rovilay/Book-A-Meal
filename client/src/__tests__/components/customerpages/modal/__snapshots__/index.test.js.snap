// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComp component test should render \`ModalComp\` correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  contentLabel="Edit Order"
  isOpen={true}
  overlayClassName="overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="closeBtn"
  >
    <button
      className="modal-btn"
      onClick={[MockFunction]}
      type="button"
    >
      <FontAwesome
        name="times"
        size="2x"
      />
    </button>
  </div>
  <OrderDetailsTable
    content={
      Object {
        "Meals": Array [
          Object {
            "OrderMeal": Object {
              "cost": 400,
              "portion": 1,
            },
            "id": 1,
            "title": "Eba and Egusi",
          },
          Object {
            "OrderMeal": Object {
              "cost": 800,
              "portion": 2,
            },
            "id": 2,
            "title": "Ewa",
          },
        ],
        "createdAt": "2018-09-01",
        "deliveryAddress": "isuti rd",
        "orderId": 6,
      }
    }
    editOrder={[MockFunction]}
    getOrderMeals={[MockFunction]}
    hideModal={[MockFunction]}
    modal={
      Object {
        "content": Object {
          "Meals": Array [
            Object {
              "OrderMeal": Object {
                "cost": 400,
                "portion": 1,
              },
              "id": 1,
              "title": "Eba and Egusi",
            },
            Object {
              "OrderMeal": Object {
                "cost": 800,
                "portion": 2,
              },
              "id": 2,
              "title": "Ewa",
            },
          ],
          "createdAt": "2018-09-01",
          "deliveryAddress": "isuti rd",
          "orderId": 6,
        },
        "contentLabel": "Edit Order",
        "isEdit": true,
        "isInfo": true,
        "isOpen": true,
      }
    }
    orderedMealsPagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    setModal={[MockFunction]}
    title="Order Details"
  />
  <EditOrderTable
    content={
      Object {
        "Meals": Array [
          Object {
            "OrderMeal": Object {
              "cost": 400,
              "portion": 1,
            },
            "id": 1,
            "title": "Eba and Egusi",
          },
          Object {
            "OrderMeal": Object {
              "cost": 800,
              "portion": 2,
            },
            "id": 2,
            "title": "Ewa",
          },
        ],
        "createdAt": "2018-09-01",
        "deliveryAddress": "isuti rd",
        "orderId": 6,
      }
    }
    editOrder={[MockFunction]}
    getOrderMeals={[MockFunction]}
    hideModal={[MockFunction]}
    isEdit={true}
    modal={
      Object {
        "content": Object {
          "Meals": Array [
            Object {
              "OrderMeal": Object {
                "cost": 400,
                "portion": 1,
              },
              "id": 1,
              "title": "Eba and Egusi",
            },
            Object {
              "OrderMeal": Object {
                "cost": 800,
                "portion": 2,
              },
              "id": 2,
              "title": "Ewa",
            },
          ],
          "createdAt": "2018-09-01",
          "deliveryAddress": "isuti rd",
          "orderId": 6,
        },
        "contentLabel": "Edit Order",
        "isEdit": true,
        "isInfo": true,
        "isOpen": true,
      }
    }
    orderedMealsPagination={
      Object {
        "count": 13,
        "limit": 12,
        "numOfPages": 2,
      }
    }
    setModal={[MockFunction]}
    title="Order Details"
  />
</Modal>
`;
